docs/src/pages/components/data-grid/components/CustomColumnMenu.js

```diff
@@ -85,29 +85,23 @@ export default function CustomColumnMenu() {
   const [color, setColor] = React.useState('primary');
   const apiRef = useGridApiRef();
 
-  React.useEffect(() => {
-    apiRef.current.showColumnMenu('default');
-  }, [apiRef, color]);
-
   return (
     <div
       style={{
-        height: 300,
         width: '100%',
-        display: 'flex',
-        flexDirection: 'column',
       }}
     >
-      <div style={{ alignSelf: 'center' }}>
-        <Button
-          color={color}
-          onClick={() =>
-            setColor((current) => (current === 'primary' ? 'secondary' : 'primary'))
-          }
-        >
-          Toggle menu background
-        </Button>
-      </div>
+      <Button
+        color={color}
+        variant="outlined"
+        onClick={(event) => {
+          event.stopPropagation();
```

Comment:
why do you need to stop propagation here? Next js issue?

Reply:
I have stopped the propagation to keep the column menu open when we toggle the button. 

AFAIK, we have never seen a rendering issue from Next.js. I don't know what would put us on this lead, it uses the public React API, so I doubt it will ever happen. I believe we can reproduce the flash on codesandbox

Reply:
No I just wondered... Maybe it is worth adding a comment.

The next js idea was total supposition.